src/RefractDataStructure.cc

```diff
@@ -801,23 +730,47 @@ namespace
 
     template <>
     struct SaveValue<EnumElement, false> {
+        using T = EnumElement;
 
-        void operator()(std::deque<ElementInfo<EnumElement> > values,
-            std::deque<ElementInfo<EnumElement> > enumerations,
-            EnumElement& element) const
+        void operator()(ElementData<T>& data, T& element, ConversionContext& context) const
         {
-            auto valuesInfo = Merge<EnumElement>()(std::move(values));
-            auto enumInfo = Merge<EnumElement>()(std::move(enumerations));
 
-            if (!valuesInfo.value.empty()) {
-                auto content = std::move(valuesInfo.value.front());
-                valuesInfo.value.pop_front();
+            if ((data.values.size() >= 1) && (data.values.front().value.size() == 1)
+                && !data.values.front().value.front()->empty()) {
+                auto content = data.values.front().value.front()->clone();
                 element.set(dsd::Enum{ std::move(content) });
             }
 
-            if (!enumInfo.value.empty()) {
-                auto enumsElement = make_element<ArrayElement>();
-                std::move(enumInfo.value.begin(), enumInfo.value.end(), std::back_inserter(enumsElement->get()));
+            auto valuesInfo = Merge<EnumElement>()(std::move(data.values));
+            auto samplesInfo = Merge<EnumElement>()(std::move(CloneElementInfoContainer<T>(data.samples)));
+            auto defaultInfo = Merge<EnumElement>()(std::move(CloneElementInfoContainer<T>(data.defaults)));
+
+            dsd::Array enums;
+
+            auto addToEnumerations = [&enums, &context](auto& info, const auto& sourceMap, const bool reportDuplicity) {
```

Comment:
Why does this lambda capture references instead of taking them as argument? Is there a reason to be un-pure?

Reply:
They are hold full time, all others param are changed, but i can move it into params